user/tests/wiring/ble/heartrate_central/application.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this program; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "application.h"
+
+BLEScanResult    results[10];
+
+BLEPeerAttribute heartrate;
+BLEPeerDevice    peer;
+
+void heartrateUpdated(uint8_t* data, size_t len) {
+
+}
+
+void setup() {
+}
+
+void loop() {
+    if (BLE.connected()) {
+        int newHr;
+        heartrate->getValue(&newHr);
+    }
+    else {
+        size_t count = BLE.scan(results, 10);

```

Comment:
This should not be `size_t`. `size_t` represents the size of an object/array/allocation in _bytes_. I assume `BLE.scan()` is returning the number of scan results, and that their size in bytes is much bigger.

Reply:
Sorry for barging in, but `size_t` is a valid type to return for the number of items. stdlib uses it for its container types (see https://en.cppreference.com/w/cpp/types/size_t).

Reply:
One suggestion here is that if `BLE.scan()` may fail, I'd advise returning a signed type (such as simply `int` or perhaps `ssize_t`) to be able to retrieve a `system_error_t`.

Reply:
Interesting! I don't actually read that cppreference.com page as saying what you're saying, but I believe you — you're by far the more knowledgeable here.

I'd agree in recommending `int` for the return type of `BLE.scan()`.